pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -900,6 +903,9 @@ func shouldStopMachines(cd *hivev1.ClusterDeployment, hibernatingCondition *hive
 // shouldStartMachines decides if machines should be started
 func shouldStartMachines(cd *hivev1.ClusterDeployment, hibernatingCondition *hivev1.ClusterDeploymentCondition,
 	readyCondition *hivev1.ClusterDeploymentCondition) bool {
+	if hibernatingCondition.Reason == hivev1.HibernatingReasonUnsupported {
+		return true
```

Comment:
Was this added to work around unsupported with OpenStack originally? If hibernation is not supported then we shouldn't start machines, correct? Comment below this indicates similar:

```go
        ...
	if hibernatingCondition.Status == corev1.ConditionFalse &&
		(hibernatingCondition.Reason != hivev1.HibernatingReasonSyncSetsApplied) {
		// reason is either ResumingOrRunning, FailedToStop, Unsupported or SyncSetsNotApplied
		return false
	}
```

Reply:
Mm, agree. I think what ends up happening as a result is we'll call `startMachines` (which we wouldn't have before) but that will immediately bail because there's no actuator (which would have been the cause of setting Unsupported in the first place). But the same thing should happen in the existing code path where we _skip_ `startMachines` and go straight to `checkClusterRunning`.

So yeah, can you explain why this change was put in?

Reply:
@abutcher , thank you for pointing this out, I can't find the discussion I had about this and I agree with you when looking at this now. IIRC hibernation controller was constantly reconciling and the cluster was never getting to "Running" state and couldn't be claimed. I think the other change in the hibernation controller is handling this.I will test again this part with Openstack.

Reply:
This is a great find @abraverm. Discussed with @abutcher yesterday in scrum. Agreed that this is a sticking point, and we need a good answer. What we came up with is: ultimately we want to force the user to accept explicitly that clusters in such pools will never hibernate. @abutcher's idea of a "switch" is a good one -- but I'd like to do it without adding another field to the API. Instead we'll force the user to set `runningCount == size` via a validating webhook.

However, let's do the minimum possible in this PR so we can land it, and we can fill in the gaps in a followon. Here's what I suggest:
Early in the clusterpool controller, add a function like (pseudocode):
```python
if pool.Spec.Platform not in (alibaba, aws, azure, gcp, ibmcloud)
    and pool.Spec.RunningCount != pool.Spec.Size:
        return reconcile.Result{}, errors.New("Hibernation not supported for $cloud clouds unless runningCount==size")
```
I.e. we'll just refuse to process a ClusterPool unless it meets those criteria; though the user won't know why it's not working unless they look at the logs.

Then in a followon PR we can add the webhook to improve the UX, and figure out how to consolidate the `hibernationSupported` function (see below).

A couple of background notes:
At the moment, the hibernation controller [decides](https://github.com/openshift/hive/blob/66d98a5ffa2935a3770348cb023b8f891184efff/pkg/controller/hibernation/hibernation_controller.go#L750-L766) whether hibernation is supported based on a ClusterDeployment's Platform and a version label. The latter is obsolete, as we [no longer support](https://access.redhat.com/support/policy/updates/openshift) versions older than [4.4.8](https://github.com/openshift/hive/blob/66d98a5ffa2935a3770348cb023b8f891184efff/pkg/controller/hibernation/hibernation_controller.go#L77-L79) (which is a relief -- we can't reasonably figure that out until partway through the installation process anyway), so we just have to filter by platform. It's not as simple as publicizing (the early part of) that function, because it operates on a CD, via actuators that are currently private to the hibernation controller. That's why the above suggestion is simply to duplicate and hardcode the list of supported platforms for the moment :(